### ISSUE TITLE
Cura "dynamic acceleration" -> does klipper does change the deceleration automaticly?

### ISSUE THREAD

**Issue by StefanMeGit** [Klipper3d/klipper#5204]
Hi, 

i recently run into a problem while cura is changing my acc on different moves (10.000 for traveling, 5000 for outer counters ect...) with the dynamic accelration function. Works like a charm, except... it looks like the deceleration is not changing accordingly to the acceleration. 

![image](https://user-images.githubusercontent.com/16327026/151359440-379ad931-79b2-4a27-babe-3faca450846c.png)
![image](https://user-images.githubusercontent.com/16327026/151359507-f300bf4f-52b6-4696-911a-454bd901a324.png)

If i set the decleration to 4000, it stays there all the time... is it possible to force the deceleration to change everythime i change the acceleration?
[klippy (6).log](https://github.com/Klipper3d/klipper/files/7950425/klippy.6.log)

**Comment by Piezoid** [comment]
No, by default M204 doesn't affect  `max_accel_to_decel`. To achieve that, you may overload M204 with a macro like this:
```property
[gcode_macro M204]
rename_existing:            M204.1
gcode:
    {% if 'S' in params %}
        SET_VELOCITY_LIMIT ACCEL={ S } ACCEL_TO_DECEL={ S|float * F|default(0.75)|float }
    {% endif %}
```
The `F` paramter controls the ratio between the two. It's default (0.75 here) can be adjusted.

Also the setting `max_accel_to_decel` *is not* deceleration. It impose a limit on peak *velocity* for short moves ([see here](https://www.klipper3d.org/Kinematics.html#smoothed-look-ahead).
